c_common/models/system_models/src/extra_monitor_support.c

```diff
@@ -1638,9 +1645,35 @@ static void data_out_initialise(void) {
     set_vic_callback(DMA_TIMEOUT_SLOT, DMA_TO_INT, data_out_dma_timeout);
 
     // configuration for the DMA's by the speed data loader
-    dma[DMA_CTRL] = 0x3f; // Abort pending and active transfers
-    dma[DMA_CTRL] = 0x0d; // clear possible transfer done and restart
-    dma[DMA_GCTL] = 0x1ffc00; // enable DMA done and error interrupt
+    dma_control->control = (dma_control_t) {
+        // Abort pending and active transfers
+        .uncommit = true,
+        .abort = true,
+        .restart = true,
+        .clear_done_int = true,
+        .clear_timeout_int = true,
+        .clear_write_buffer_int = true
+    };
+    dma_control->control = (dma_control_t) {
+        // clear possible transfer done and restart
+        .uncommit = true,
+        .restart = true,
+        .clear_done_int = true
+    };
+    dma_control->global_control = (dma_global_control_t) {
+        // enable DMA done and error interrupt
+        .transfer_done_interrupt = true,
+        .transfer2_done_interrupt = true,
+        .timeout_interrupt = true,
+        .crc_error_interrupt = true,
+        .tcm_error_interrupt = true,
+        .axi_error_interrupt = true, // SDRAM error
+        .user_abort_interrupt = true,
+        .soft_reset_interrupt = true,
+        .unknown1 = true, // ???
```

Comment:
whats with the unknowns?

Reply:
why are we setting them to true? what do these do?

Reply:
They say that if the corresponding signal is set in another register, an interrupt should be raised. But those bits in that other register are unassigned. Whee! 🤦 

Removed because then I get (slightly) nicer documentation.